mpl3115a2.cpp

```diff
@@ -64,3 +61,38 @@ MPL3115A2::MPL3115A2(unsigned int adapterNumber)
                   << m_i2cFilename << std::endl;
     }
 }
+
+
+std::vector<uint8_t> MPL3115A2::readBytes(const uint8_t reg, const unsigned int size) const
+{
+    std::vector<uint8_t> regVector {reg};
+    writeBytes(regVector);  // Tell slave where the data we want is
+    std::unique_ptr<uint8_t> cArray(new uint8_t[size]);
+    if (read(m_i2cFile, cArray.get(), 1) != 1)
+    {
+        std::ostringstream err;
+        err << "Could not read from MPL3115A2" << strerror(errno);
+        throw std::runtime_error(err.str());
+    }
+    std::vector<uint8_t> data;
+    data.assign(cArray.get(), cArray.get() + size);
+    return data;
+}
+
+
+void MPL3115A2::writeBytes(const std::vector<uint8_t> &data) const
+{
+    std::unique_ptr<uint8_t> cArray(new uint8_t[data.size()]);
+    uint8_t *currentElement = cArray.get();
+    for (uint8_t x : data)
```

Comment:
got rid of the auto call, just more confusing in this situation than it is helpful